Fix:

dap0.c:241:3: error: implicit declaration of function 'dap_main' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
ps1.c:904:29: error: implicit declaration of function 'dap_il' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
ps1.c:905:29: error: implicit declaration of function 'dap_sl' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
ps1.c:906:30: error: implicit declaration of function 'dap_dl' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
sbstrans.c:214:7: error: implicit declaration of function 'unget1c' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
sbstrans.c:1268:5: error: implicit declaration of function 'importtrans' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
sbstrans.c:1270:5: error: implicit declaration of function 'surveyselecttrans' is invalid in C99 [-Werror,-Wimplicit-function-declaration]

https://mail.gnu.org/archive/html/bug-dap/2021-01/msg00000.html
--- src/dap0.c.orig	2014-04-16 13:36:34.000000000 -0500
+++ src/dap0.c	2021-01-06 08:45:53.000000000 -0600
@@ -91,6 +91,8 @@
 static int nmallocs = 0;
 static int nfrees = 0;
 
+extern int dap_main(int argc, char **argv);
+
 char *dap_malloc(int nbytes, char *mesg)
 {
   char *m;
--- src/ps1.c.orig	2014-04-16 13:36:34.000000000 -0500
+++ src/ps1.c	2021-01-06 08:41:30.000000000 -0600
@@ -34,6 +34,9 @@
 extern dataobs dap_obs[];
 extern FILE *dap_err;
 
+extern void dap_dl(char varname[], double *dbl);
+extern void dap_il(char varname[], int *i);
+extern void dap_sl(char varname[], char *s);
 extern char *pict_newstr(char *str);
 
 void pict_maketick(tick *t, double num, char *label, double len)
--- src/sbstrans.c.orig	2014-04-16 13:36:34.000000000 -0500
+++ src/sbstrans.c	2021-01-06 08:44:04.000000000 -0600
@@ -53,6 +53,10 @@
   ""
 };
 
+extern void importtrans(char *step, FILE *dapfile);
+extern void surveyselecttrans(char *step, FILE *dapfile);
+extern void unget1c(int c, FILE *dotc, FILE *dapc);
+
 int iskeyword(char *str)
 {
   int k;
